lib/ansible/plugins/strategy/linear.py

```diff
@@ -213,7 +213,7 @@ def run(self, iterator, play_context):
 
                         run_once = templar.template(task.run_once) or action and getattr(action, 'BYPASS_HOST_LOOP', False)
 
-                        if (task.any_errors_fatal or run_once) and not task.ignore_errors:
+                        if task.any_errors_fatal and not task.ignore_errors:
```

Comment:
I wonder if the second paragraph from my first comment above applies here in some way too because here we declare that `run_once` is equivalent to `templar.template(task.run_once) or action and getattr(action, 'BYPASS_HOST_LOOP', False)` but in `StrategyBase._process_pending_results()` we check only against `original_task.run_once`.

Both of my comments however make me wonder in what scenarios is the code to handle `run_once` in `StrategyBase._process_pending_results()` even needed since it is currently handled, it seems, by the `any_errors_fatal` code.

Note that the code for `run_once` and `any_errors_fatal` are similar but not the same, it appears to be different for unreachable hosts - https://github.com/ansible/ansible/pull/82852.

Reply:
So I think this is correct change but as I mentioned above it would break the following playbook:

```yaml
- hosts: h1,h2
  gather_facts: false
  tasks:
    - name: missing task args, this fails
      add_host:

    - debug:
        msg: add_host is BYPASS_HOST_LOOP action and implies run_once, this should not happen
```

because we don't check for `BYPASS_HOST_LOOP` in `StrategyBase` to fail all other hosts, we just check for `run_once`. So I think that `BYPASS_HOST_LOOP` check needs to be somehow added there.